### DISCUSSION TITLE
Add "boot count" to tele STATE JSON

### DISCUSSION THREAD

**Discussion by ortegafernando** [arendst/Tasmota#14747] [Ideas]
Hi, do you think that bootcount is important to detect failures, or bugs, or power source problems, or .... ?

Could be a good idea to include "bootCount" to tele STATE JSON to monitorize it and make thresholds, alarms, and so on?

Another option could be to add it to INFO3 JSON, but I think is better to get it in STATE tele JSON.

What is your opinion? Thanks.

**Comment by s-hadinger** [comment]
Boot count does not change very often except at reboots (obviously) so there is no point in adding it to teleperiod. 

You can get this information with `Status 1`. You can also detect a reboot by the pattern of received messages.

**Reply by ortegafernando** [reply]
But we can also have reboots without connecting to wifi or even connecting to MQTT.

I know about STATUS 1, but it implies to send it once a ... 

I understand that "normally" it doesn't change often. But the problem here is to detect abnormal working of tasmota.

If people dont like it in TELE STATE, it could be in INFO3, so we only recive it once, everytime tasmota connects to MQTT broker.

May be INFO3 is not a bad idea.

**Reply by s-hadinger** [reply]
I understand but bear in mind that adding this would impact everybody. If it's only for debugging or special circumstances it should be reserved for such situations. 

You can easily add a rule to trigger a Status 1 command at each Mqtt connection

**Reply by sfromis** [reply]
I'm having a small automation doing a `Status0` after each boot, available to anyone listening, including my MQTT logger.

**Reply by ortegafernando** [reply]
> I understand but bear in mind that adding this would impact everybody. If it's only for debugging or special circumstances it should be reserved for such situations.
> 
> You can easily add a rule to trigger a Status 1 command at each Mqtt connection

Excuse, why woud it impact? As it is a JSON message, it will be only one new object inside JSON, so i don't know why it would affect everybody.

Instead of:
```
{"Info3":{"RestartReason":"Software/System restart"}}
```

It could be:
```
{"Info3":{"RestartReason":"Software/System restart","BootCount":"45"}}
```

All your program's logic (nodered for example) will not be affected, will it?

Thanks

**Reply by ortegafernando** [reply]
> I'm having a small automation doing a `Status0` after each boot, available to anyone listening, including my MQTT logger.

Thanks, it could be an option thanks. But I think that incluide this value in INFO3 it is not risky for anyone. Thanks for this great possibility.

**Reply by arendst** [reply]
This is the nicest issue:
```
18:06:00.686 MQT: tele/esp32d/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":442}}
18:06:00.709 TFS: File 'autoexec.bat' not found
18:06:02.788 QPC: Reset
18:06:04.768 APP: Boot Count 443
```
To solve possible flash corruption the bootcount is updated AFTER the Info3 message is presented :-) as it initiates a flash write.

**Reply by arendst** [reply]
Just merged in development branch with an additional one as correction ;-)

**Reply by ortegafernando** [reply]
> Just merged in development branch with an additional one as correction ;-)

Sorry @arendst , what about "Flash write Count". Could it be possible to add to INFO3 ??

May be a timer or ruleTimer is bad configurated, or with a bad "var" value and it is flashing continuosly.

With these two values we could detect/log/monitorize any strange problem.
